client/verta/verta/deployment/_deployedmodel.py

```diff
@@ -132,46 +111,63 @@ def prediction_url(self, value):
             raise ValueError("not a valid prediction_url")
         self._prediction_url = parsed.geturl()
 
-    @property
-    def credentials(self):
-        return self._credentials
 
-    @credentials.setter
-    def credentials(self, value):
-        self._credentials = value
-        self._init_session()
+    # TODO: Implement dynamic compression via separate utility and call it from here
+    def _predict(
+            self,
+            x: Any,
+            compress: bool=False,
+            prediction_id: Optional[str]=None,
+            ):
+        """This is like ``DeployedModel.predict()``, but returns the raw ``Response``
+         for debugging."""
+        request_headers = dict()
+        if prediction_id:
+            request_headers.update({'verta-request-id': prediction_id})
 
-    @property
-    def access_token(self):
-        return self._access_token
-
-    @access_token.setter
-    def access_token(self, value):
-        self._access_token = value
-        self._init_session()
-
-    def _predict(self, x, compress=False):
-        """This is like ``DeployedModel.predict()``, but returns the raw ``Response`` for debugging."""
         x = _utils.to_builtin(x)
         if compress:
+            request_headers.update({'Content-Encoding': 'gzip'})
             # create gzip
             gzstream = six.BytesIO()
             with gzip.GzipFile(fileobj=gzstream, mode="wb") as gzf:
                 gzf.write(six.ensure_binary(json.dumps(x)))
             gzstream.seek(0)
 
-            return self._session.post(
+            response = self._session.post(
                 self._prediction_url,
-                headers={"Content-Encoding": "gzip"},
+                headers=request_headers,
                 data=gzstream.read(),
-            )
+                )
         else:
-            return self._session.post(self.prediction_url, json=x)
+            response = self._session.post(
+                self.prediction_url,
+                headers=request_headers,
+                json=x,
+                )
+        if response.status_code in (
+            400,
+            502,
+            ):  # possibly error from the model back end
+            try:
+                data = _utils.body_to_json(response)
+            except ValueError:  # not JSON response; 502 not from model back end
+                pass
+            else:  # from model back end; contains message (maybe)
+                # try to directly print message, otherwise line breaks appear as '\n'
+                msg = data.get("message") or json.dumps(data)
+                raise RuntimeError(f"Deployed model encountered an error: {msg}")
+
+        elif not response.ok:
```

Comment:
I think this should be

```suggestion
        if not response.ok:
```

to catch the

```python
except ValueError
    pass
```

from above (and I think it's safe to be able to assume that `_predict()` will always return a successful `response`, and catch + raise errors otherwise)